drivers/ds3231/ds3231.c

```diff
@@ -236,6 +257,158 @@ int ds3231_set_time(const ds3231_t *dev, const struct tm *time)
     return 0;
 }
 
+#if IS_USED(MODULE_DS3231_INT)
+int ds3231_await_alarm(ds3231_t *dev)
+{
+    mutex_t mutex = MUTEX_INIT_LOCKED;
+
+    assert(dev != NULL);
+    assert(gpio_is_valid(dev->int_pin));
+
+    if (gpio_init_int(dev->int_pin, GPIO_IN, GPIO_FALLING, _unlock, &mutex) < 0) {
+            return -EIO;
+    }
+
+    /* wait for alarm */
+    mutex_lock(&mutex);
+
```

Comment:
You should also cal `gpio_irq_disable` here, otherwise if the interrupt happens a second time (when outside this function) `&mutex` will point to random stack memory.

Reply:
Good catch. But quick question, does `gpio_init_int` always does some sort of a `gpio_irq_enable` (it looks like it but I'm not sure) ? Otherwise next time `ds3231_await_alarm` is called, the GPIO irq will be disabled if, I'm not mistaken.